library/scripts/adi_make_lib.tcl

```diff
@@ -0,0 +1,189 @@
+  set msg_level 0
+  ##############################################################################
+
+  # global variables
+  variable library_dir
+  variable done_list
+  variable serch_pattern
+
+  # init local namespace variables
+  set done_list ""
+  set build_list ""
+  set makefiles ""
+  set depend_lib ""
+  set match ""
+
+  # searching for subdir libraries in path for first argument
+  set first_lib [lindex $argv 0]
+  if { $first_lib == "" } {
+    set first_lib "."
+  }
+
+  # get library absolute path
+  set glb_path [file normalize $first_lib]
+  if { [regexp library $glb_path] } {
+    regsub {library.*$} $glb_path library library_dir
+  } else {
+    puts "ERROR: Not in library/* folder or argument to an IP located in library folder"
+    return
+  }
+
+  # geting all parsed arguments (libraries)
+  set index 0
+  set library_element(1) $first_lib
+  foreach argument $argv {
+    incr index 1
+    set library_element($index) $argument
+  }
+
+  # definie library dependancy search (Makefiles)
+  set serch_pattern "XILINX_.*_DEPS.*="
+
+  #############################################################################
+  # library build procedures
+  #############################################################################
+
+  # have verbous or debug messages ############################################
+  proc puts_msg_level { message } {
+    global msg_level
+    if { $msg_level == 1 } {
+      puts $message
+    }
+  }
+
+  # search for IP dependancies #################################################
+  proc search_dependancy { path } {
+
+    puts_msg_level "DEBUG search_dependancy proc"
+    global serch_pattern
+    global library_dir
+    set match ""
+    set fp1 [open $library_dir/$path/Makefile r]
+    set file_data [read $fp1]
+    close $fp1
+
+    set lines [split $file_data \n]
+    foreach line $lines {
+      regexp $serch_pattern $line match
+      if { $match != "" } {
+        regsub -all $serch_pattern $line "" lib_dep
+        set lib_dep [string trim $lib_dep]
+	puts_msg_level "    > dependancy library $lib_dep"
+	# build dependancy
+        build_lib $lib_dep
+	set match ""
+      }
+    }
+  }
+
+  # build procedure ############################################################
+  proc build_lib { library } {
+
+    puts_msg_level "DEBUG build_lib proc"
+    global done_list
+    global library_dir
+
+    # determine if the IP was previously build in the current adi_make_lib.tcl call
+    if { [regexp $library $done_list] } {
+      puts_msg_level "Build previously done on $library"
+      return
+    } else {
+      puts_msg_level "- Start build of $library"
+    }
+    puts_msg_level "- Search dependancies for $library"
+
+    # search for current IP dependancies
+    search_dependancy $library
+
+    puts_msg_level "- Continue build on $library"
+    set ip_name "[file tail $library]_ip"
+
+    cd $library_dir/${library}
+    exec vivado -mode batch -source "$library_dir/${library}/${ip_name}.tcl"
```

Comment:
You can't open a project (IP) if you have another project open, (e.g. daq2_zc706).
There are more workarounds... 
https://github.com/analogdevicesinc/hdl/blob/adi_make_scripts/projects/scripts/adi_make.tcl#L135-L136

I agree that this will look/work better in a procedure in some namespace. At the time, I created a script to build the libraries and needed a mode to call it with some arguments. After that just added more code on top.
I will look into it.

Reply:
I think in general you not going to have any opened project, you just start to build one. You should have a clean Vivado, otherwise you should close your current project.